### PR TITLE
Add 'xxhash'

### PR DESCRIPTION
**Pull request by Warchant**
Signed-off-by: Bogdan Vaneev <warchantua@gmail.com>

* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/create/cmake.html)
  step by step carefully. **[Yes]**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/Warchant/hunter/builds/22368298
  * https://travis-ci.com/Warchant/hunter/builds/100927262

* I have submitted CI configs to https://github.com/ingenue/hunter targeting `pkg.template` branch,
  see this merged pull request https://github.com/ingenue/hunter/pull/334

---

P.S. As package migrated to hunter-packages, I updated version and CI travis/appveryor.

### CODE REVIEW

File: cmake/configs/default.cmake [hunk 1]
```diff
@@ -457,6 +457,7 @@ hunter_default_version(xrender VERSION 0.9.7)
 hunter_default_version(xshmfence VERSION 1.3)
 hunter_default_version(xtrans VERSION 1.2.7)
 hunter_default_version(xxf86vm VERSION 1.1.2)
+hunter_default_version(xxhash VERSION v0.6.5-p1)
```
Comment (ruslo) [inline cmake/configs/default.cmake]: Usually, in this file it's without `v`, can you change `v0.6.5-p1` to `0.6.5-p1` please?

File: cmake/projects/xxhash/hunter.cmake [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+include(hunter_add_version)
+include(hunter_cacheable)
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME
+    xxhash
+    VERSION
+    v0.6.5-p1
```
Comment (ruslo) [inline cmake/projects/xxhash/hunter.cmake]: `0.6.5-p0`

File: docs/packages/pkg/xxhash.rst [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+.. index::
+  single: crypto ; xxhash
+
+.. _pkg.xxhash:
+
+xxhash
+======
+
+-  `Official <https://github.com/Cyan4973/xxHash>`__
+-  `Hunterized <https://github.com/Warchant/xxHash>`__
```
Comment (ruslo) [inline docs/packages/pkg/xxhash.rst]: Should we move it to https://github.com/hunter-packages ?
- https://docs.hunter.sh/en/latest/creating-new/patch.html

Comment (Warchant) [reply docs/packages/pkg/xxhash.rst]: Yes. Should I transfer it to hunter-packages?

Comment (ruslo) [reply docs/packages/pkg/xxhash.rst]: Created: https://github.com/hunter-packages/xxHash

Please send your patches there. Also, it's better to ask for it before submitting a pull request, see:
- https://docs.hunter.sh/en/latest/creating-new/patch.html

**Comment by Warchant** [comment]
@ruslo bump. Repo migrated, CI passed.

**Comment by ruslo** [comment]
Testing:
- https://ci.appveyor.com/project/ingenue/hunter/builds/22419415
- https://travis-ci.org/ingenue/hunter/builds/494162237?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
- https://ci.appveyor.com/project/ingenue/hunter/builds/22419707
- https://travis-ci.org/ingenue/hunter/builds/494169150?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.106

**Pull request merged**